### PR TITLE
Add kernel configuration for Cisco platforms

### PR DESCRIPTION
**Pull request by madhava28**
Kernel configuration is enabled to support device drivers for sensor devices,
gpio devices, MDIO mux devices, ADC, DAC device and Intel SPI controller.

Drivers are enabled for all cisco-8000 platforms.

New image with updated kernel configs will increase size of Sonic image by 40Kb

Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>

Initial PR link: https://github.com/sonic-net/sonic-linux-kernel/pull/301

Testing Done:
Attaching 
[Sonic_bootlogs.txt](https://github.com/sonic-net/sonic-linux-kernel/files/10115688/Sonic_bootlogs.txt)
Sonic boot logs with Cisco kernel configuration enabled.

### CODE REVIEW

**Comment by bmridul** [comment]
@kevinskwang @sachinv-msft , pls help review this PR.

**Review by kevinskwang — APPROVED** [review]

**Comment by kevinskwang** [comment]
@madhava28 Please authorize the EasyCLA to make it pass.

File: patch/kconfig-inclusions [hunk 1]
```diff
@@ -81,6 +81,85 @@ CONFIG_SENSORS_EMC2305=m
...
+CONFIG_AD7303=m
+CONFIG_AD8801=m
+CONFIG_MAX517=m
+CONFIG_MCP4725=m
+CONFIG_MCP4922=m
+CONFIG_TI_DAC082S085=m
+CONFIG_TI_DAC5571=m
+CONFIG_SPI_INTEL_SPI_PLATFORM=m
+CONFIG_SPI_INTEL_SPI=m
+CONFIG_SERIAL_8250_RUNTIME_UARTS=4
```
Comment (saiarcot895) [inline patch/kconfig-inclusions:162]: The default value for `CONFIG_SERIAL_8250_RUNTIME_UARTS` is already 4, does it need to be re-specified here?

Comment (madhava28) [reply patch/kconfig-inclusions:162]: Hi Saikrishna,
I set it as 4 explicitly to keep track of value. Eventually we plan to remove this and make it as part of bootargs

File: patch/kconfig-inclusions [hunk 2]
```diff
@@ -81,6 +81,85 @@ CONFIG_SENSORS_EMC2305=m
...
+CONFIG_AD5755=m
+CONFIG_AD5758=m
+CONFIG_AD5761=m
+CONFIG_AD5764=m
+CONFIG_AD5791=m
+CONFIG_AD7303=m
+CONFIG_AD8801=m
+CONFIG_MAX517=m
+CONFIG_MCP4725=m
+CONFIG_MCP4922=m
```
[the remark this answers is not in the document]
Comment (madhava28) [reply patch/kconfig-inclusions]: Removed all the duplicate entires from the kconfig-inclusion file which are part for default config file

File: patch/kconfig-inclusions (hunk 1 above)
Comment (madhava28) [reply patch/kconfig-inclusions]: I have removed this as well

**Review by saiarcot895 — APPROVED** [review]

**Pull request merged**